### DISCUSSION TITLE
Is there a smart way to deal with SwiftLint warnings in WithViewStore?

### DISCUSSION THREAD

**Discussion by masurao3** [pointfreeco/swift-composable-architecture#2377] [Q&A] [answered]
When I use the new WithViewStore, I get the following warning when I use SwiftLint.

```
WithViewStore(self.store, observe: { $0 }) { viewStore in
```

```Multiple Closures with Trailing Closure Violation: Trailing closure syntax should not be used when passing more than one closure argument (multiple_closures_with_trailing_closure)```

Although I can change the SwiftList settings to avoid the warning, but I would like to solve the warnings by changing the way WithViewStore is written if possible. Is there any way to do this?

**Comment by pyrtsa** [comment, marked as answer]
To that very use case, you could define
```swift
func identity<T>(_ value: T) -> T { value }
```
and replace `observe: { $0 }` with `observe: identity`. But it won't scale if you need to do anything else that can't be a simple key path expression.

(Curiously, `observe: \.self` also doesn't work because of a limitation in the Swift compiler.)

**Reply by masurao3** [reply]
The code looks simple and warnings have been fixed! Thank you!

**Comment by mbrandonw** [comment]
Another alternative is to use multiple trailing closures:

```swift
WithViewStore(self.store) {
  $0
} content: { viewStore in
  …
}
```

But it does seem like a kinda silly SwiftLint rule.

**Reply by masurao3** [reply]
I thought this idea looked good as well. Thank you!
